### PR TITLE
Add support for network host mode and pass dns to the docker

### PR DESCRIPTION
**Pull request by zmalik**
-support for the network mode host
-pass the dns parameter to the docker

As we discussed earlier I have added the flag for the dns to avoid opening the parameters in the Eremetic API.

### CODE REVIEW

File: mesos/task.go [hunk 1]
```diff
@@ -45,6 +48,34 @@ func createTaskInfo(task eremetic.Task, offer *mesosproto.Offer) (eremetic.Task,
...
+	for k, v := range params {
+		parameters = append(parameters, &mesosproto.Parameter{
+			Key: proto.String(k),
+			Value: proto.String(v),
+		})
+	}
+	return parameters
+}
+
+func buildNetwork(task eremetic.Task) *mesosproto.ContainerInfo_DockerInfo_Network {
```
Comment (keis) [inline mesos/task.go]: This looks a very round about way of doing it. I think this (untested) should do the same

```
func buildNetwork(task eremetic.Task) *mesosproto.ContainerInfo_DockerInfo_Network {
    if task.Network == "" {
        return mesosproto.ContainerInfo_DockerInfo_BRIDGE.Enum()
    }
    return mesosproto.ContainerInfo_DockerInfo_Network(mesosproto.ContainerInfo_DockerInfo_Network_value[task.Network]).Enum()
}
```

File: mesos/task.go [hunk 2]
```diff
@@ -45,6 +48,34 @@ func createTaskInfo(task eremetic.Task, offer *mesosproto.Offer) (eremetic.Task,
...
+func buildDockerCliParameters(task eremetic.Task) []*mesosproto.Parameter {
+	//To be able to move away from docker CLI in future, parameters aren't fully exposed to the API
+	params := make(map[string]string)
+	if task.DNS != "" {
+		params["dns"] = task.DNS
+	}
+	var parameters []*mesosproto.Parameter
+	for k, v := range params {
+		parameters = append(parameters, &mesosproto.Parameter{
+			Key: proto.String(k),
```
Comment (keis) [inline mesos/task.go]: This should be indented so please make sure to run gofmt on the file :)

**Comment by vvsh** [comment]
@zmalik, if you have some time, could you please also add ability to pass "net" parameter with a custom network name (add new parameter to eremetic.Request)?

File: mesos/task.go (hunk 1 above)
Comment (zmalik) [inline mesos/task.go]: oh yep, thanks! let me check it

**Comment by zmalik** [comment]
@vvsh you mean network-alias? with network isn't sufficient for you?

**Comment by vvsh** [comment]
@zmalik, I need to run a container in a specific network that is created by docker compose.

E.g. I can do it using the following comand:
docker run -it --link redis:redis --net=deployment_default busybox

Where "deployment_default" is name of the docker-compose network.

**Comment by keis** [comment]
Merging this as it solves the basic use-case of selecting between bridge/host mode. I think to use named networks we should populate the NetworkInfo struct (http://mesos.readthedocs.io/en/latest/networking/)

**Comment by zmalik** [comment]
@vvsh ok got it.

The shortcut can be the --net support as docker cli. 
But the right way of dealing with this is, adding support for network mode "USER". 

And quoting from mesos site [link](http://mesos.apache.org/documentation/latest/networking/)

> For NONE, HOST and BRIDGE network mode the framework only needs to specify the network mode in the DockerInfo protobuf. However, for the USER mode, since a user-defined docker network is identified by a canonical network name (similar to CNI networks) apart from setting the network mode in DockerInfo the framework also needs to specify the name field in the NetworkInfo protobuf corresponding to the name of the user-defined docker network

So that means we need two changes in Eremetic, support for the **canonical network name** and update the mesos-go vendor as the version right now in Eremetic, doesn't support the USER mode.

That can be easily another PR, where the most work is making sure we aren't going to break anything.

**Pull request merged**